uast/npath.go

```diff
@@ -0,0 +1,225 @@
+package uast
+
+import (
+	"errors"
+)
+
+//CodeReference
+//https://pmd.github.io/pmd-5.7.0/pmd-java/xref/net/sourceforge/pmd/lang/java/rule/codesize/NPathComplexityRule.html
+
+//NpathComplexity return a slice of int for each function in the tree
+func NpathComplexity(n *Node) ([]int, error) {
+	var funcs []*Node
+	var npath []int
+
+	if n.containsRol(FunctionDeclarationBody) {
+		funcs = append(funcs, n)
+	} else {
+		funcs = n.deepChildrenOfRole(FunctionDeclarationBody)
+	}
+	if len(funcs) <= 0 {
+		npath = append(npath, -1)
+		return npath, errors.New("Function declaration not found")
```

Comment:
Yeah, the cyclomatic just doesn't check that it's a function because Python, Nim, etc can have freestanding code, some other languages doesn't even have functions (brainfuck et all), etc.